script.js

```diff
@@ -1,12 +1,130 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
-}
+let body = document.querySelector('body');
+const rootElem = document.getElementById("root"); // main div with all episodes
+  let allEpisodes = getAllEpisodes();  //get data from function
+  let numOfEp = allEpisodes.length;
+
+  let select = document.createElement('select'); // create dropdown 
+  select.id= "dropdwn";
+  body.insertBefore(select, rootElem);
+
+  let selectHolder = document.createElement('option'); // create options for dropdown
+  selectHolder.value ="";
+  selectHolder.innerText ="Select Episode";
+  select.appendChild(selectHolder);
+
+let input = document.createElement("input"); //create input field
+   input.type="search";
+   input.id="search";
+   input.placeholder="Search...";
+   body.insertBefore(input, rootElem);
+   const search = document.getElementById('search');
+
+let displayNum = document.createElement('div');  //how many episodes on the screen
+displayNum.id = "displayNum";
+   body.insertBefore(displayNum, rootElem);
+   displayNum.innerText = `displaying ${allEpisodes.length}/${allEpisodes.length} episodes.`;
+
+let dataFrom = document.createElement('p'); // element telling where data is from
+dataFrom.innerHTML = `This data is from <a href =${"https://www.tvmaze.com/shows/82/game-of-thrones"} >TVMaze.com</a> `;
+body.appendChild(dataFrom);
+
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+let arr =[];   //Creating array of data needed
+
+ for(let i=0; i<numOfEp; i++){
+  // Zero pads to two digits for season and episode.
+     prepended_out = ("0" + allEpisodes[i].season ).slice(-2);
+   prepended_out2 = ("0" + allEpisodes[i].number ).slice(-2);
+let epData = {
+            name:`${allEpisodes[i].name}`,
+           season: `${prepended_out}`,
+           number: `${prepended_out2}`,
+           image: `${allEpisodes[i].image.medium}`,
+           summary: `${allEpisodes[i].summary}`,
+           id: `${allEpisodes[i].id}`
+          }; 
+   arr.push(epData);
 }
 
-window.onload = setup;
+
+//episode data to be displayed on screen
+function display(){
+  let data = arr.map((ep)=> {
+    prepended_out = ("0" + ep.season ).slice(-2);
```

Comment:
Investigating slice(-2) here. Maybe an opportunity for me to learn too.  Would ("0" + ep.season) not already be in format you need e.g. 02 or am I missing something. To me this reads that you make 02, then you cut it out and assign it to prepended out.

Reply:
I cant remember the exact explanation I found while I was researching to do that.. but basically, ("0" + ep.season), concatenates the 0 I need and the slice slices the last two digits so like.. if its a number greater or equal to 10.. those last two digits will show but less than will add the concatenated 0.. Does that make sense?